DietPi-Drive_Manager | Fix issues with autofs drives

MichaIng opened the pull request:
**Status**: Ready

**Commit list/description**:
+ DietPi-Drive_Manager | Trigger automounts to assure those are detected and shown as mounted
+ DietPi-Drive_Manager | Use lsblk to detect UUIDs for mounted drives as well, as findmnt prints an empty extra line for autofs entries

MichaIng committed:
v7.2

+ DietPi-Drive_Manager | Trigger automounts to assure those are detected and shown as mounted
+ DietPi-Drive_Manager | Use lsblk to detect UUIDs for mounted drives as well, as findmnt prints an empty extra line for autofs entries

Joulinar commented:
not sure how I did it but I found 2 USB stick at home having same `PARTUUID`. Probably I do to much testing 🤣 

```
root@DietPi4:~# lsblk -o name,fstype,label,size,ro,type,mountpoint,partuuid,uuid
NAME   FSTYPE LABEL   SIZE RO TYPE MOUNTPOINT PARTUUID                             UUID
sda                  14.3G  0 disk
├─sda1 vfat   boot    256M  0 part /boot      8f4dbd00-01                          4ADC-240F
└─sda2 ext4   rootfs 14.1G  0 part /          8f4dbd00-02                          89cef7ba-d45a-47e7-8d42-aff18afcec80
sdb                   3.8G  0 disk
└─sdb1 vfat   DEMO    3.8G  0 part            8f4dbd00-01                          3235-7350
root@DietPi4:~#
```

Ok fixed, back to the original question 😃 

---

After a reboot, my test drive is not mounted

```
root@DietPi4:~# df -h
Filesystem      Size  Used Avail Use% Mounted on
/dev/root        14G  1.1G   13G   9% /
devtmpfs        3.9G     0  3.9G   0% /dev
tmpfs           3.9G     0  3.9G   0% /dev/shm
tmpfs           3.9G   17M  3.9G   1% /run
tmpfs           5.0M     0  5.0M   0% /run/lock
tmpfs           3.9G     0  3.9G   0% /sys/fs/cgroup
tmpfs            50M  8.0K   50M   1% /var/log
tmpfs           3.9G     0  3.9G   0% /tmp
/dev/sda1       253M   32M  222M  13% /boot
root@DietPi4:~#
```

Opening `dietpi-drive_manager` will show them mounted as expected 

```
root@DietPi4:~# dietpi-drive_manager
[ INFO ] DietPi-Drive_Manager | Detecting drives, please wait...
[ INFO ] DietPi-Drive_Manager |  - Detected mounted drive: /dev/sda2 > /
[ INFO ] DietPi-Drive_Manager |  - Detected mounted drive: /dev/sda1 > /boot
[ INFO ] DietPi-Drive_Manager |  - Detected mounted drive: /dev/sdb1 > /mnt/demo
[  OK  ] DietPi-Drive_Manager | mv .fstab /etc/fstab
[  OK  ] DietPi-Drive_Manager | systemctl daemon-reload
[  OK  ] DietPi-Drive_Manager | sync
```

```
┌───────────────────────────────────────────────┤ DietPi-Drive_Manager ├───────────────────────────────────────────────┐
│ Please select a drive to see available options.                                                                      │
│  - User data location: RootFS (/mnt/dietpi_userdata)                                                                 │
│                                                                                                                      │
│                                    ●─ sda ────────────────────────────────────────────────────────                   │
│                  /                 : /dev/sda2 | ext4 | Capacity: 13.8G | Used: 1.1G (8%)                            │
│                  /boot             : /dev/sda1 | vfat | Capacity: 252M | Used: 31M (12%)                             │
│                                    ●─ sdb ────────────────────────────────────────────────────────                   │
│                  /mnt/demo         : /dev/sdb1 | ext4 | Capacity: 3.6G                                               │
│                                    ●─ Global Options ─────────────────────────────────────────────●                  │
│                  Idle Spindown     : Set a global idle duration, before drives power down                            │
│                                    ●─ Add / Refresh Drives ───────────────────────────────────────●                  │
│                  Add network drive : Select to mount networked drives                                                │
│                  Refresh           : Scan for recently added/removed drives                                          │
│                                                                                                                      │
│                                                                                                                      │
│                                  <Ok>                                      <Exit>                                    │
│                                                                                                                      │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

MichaIng committed:
v7.2

+ DietPi-Patch | Check for filesystem types via blkid, since lsblk does not work in containers

MichaIng committed:
v7.2

+ DietPi-Drive_Manager | Every drive that is mounted, has a filesystem, otherwise it would be impossible to mount it
+ DietPi-Drive_Manager | Assure that findmnt will never print two lines. Use "-U" option to omit non-unique overmounted or parent entries, like autofs lines for every drive with an x-systemd.automount unit. Use "-f" to print only the first remaining line, which covers bind mounts who always have two lines. Also be more explicit about whether the mount source or mount target is given to findmnt, as it can generally handle both.
+ DietPi-Drive_Manager | Network drives have no UUID, no partitions and hence no PARTUUID, so obtain those only for physical drives
+ DietPi-Drive_Manager | Network drives have a filesystem type, that indicates the network drive type, hence obtain and show it. But since blkid only works for physical devices and cannot take network drive sources which contain IPs, hostnames and such, findmnt must be used here. This is okay as findmnt was only found to cause issues with NTFS, showing fuseblk, which cannot be added to fstab instead of ntfs. As network drives are not added to fstab anyway, the filesytem type is purely visual here and cannot cause issues.
+ DietPi-Drive_Manager | Be more explicit about not adding network drives to fstab. All known network drive types are handled prior to the mounted loop and in every other case we do not know how the fstab entry needs to look like. To be failsafe, still check for UUID and print an error + skip the fstab entry, which strictly requires the UUID as mount source. For easier debugging, during detection loop, print info whether the mounted drive was detected as physical or as network drive.
+ DietPi-Drive_Manager | Do not use lsblk to obtain UUID and PARTUUID for unmounted drives, as this does not work in containers. lsblk can only be used for device name/path, parent device name/path, mount point and size, from what we know and is relevant for us. Use blkid for unmounted drives and findmnt for mounted drive as replacement.
+ DietPi-Drive_Manager | When unmounting a drive, assure first that the automount unit has been stopped, as it can prevent the mount point directory from being removed

MichaIng committed:
v7.2

+ DietPi-Drive_Manager | Harden one more findmnt call

MichaIng committed:
v7.2

+ CHANGELOG | Resolved an issue where info for drives with a systemd.automount unit was not detected correctly, leading to invalid fstab entries

Joulinar commented:
did a brief test, seems to be fine from my side.
